### ISSUE TITLE
hotw to estimate gpu memory needed for knowledge distillation? 

### ISSUE THREAD

**Issue by chuangzhidan** [pytorch/torchtune#2213]
its not a bug ,just have a question . tecaher  model is a14b-sized model ,student model is 3b. how much gpu memory needed for knowledge distillation? 
thank you !!

**Comment by ebsmothers** [comment]
Hi @chuangzhidan thanks for creating the issue. The answer will also depend on what you're using for an optimizer, but in general you will need model weights + gradients + optimizer state + activations for the student model, and model weights only for the teacher model. Assuming everything is in bf16 that's 2 bytes per param. So 6 GiB in weights for the student model, another 6 GiB in gradients, 12 GiB in optimizer states (assuming AdamW in bf16), then 28 GiB in model weights for the teacher model. Activations depends and is hard to estimate, but adding those up you should need at least 52 GiB for a naive implementation. This is just a rough approximation btw, you can also get peak memory stats from our logger (or enable the memory profiler if you want even more detailed memory allocations)

**Comment by joecummings** [comment]
> thank you for your help , really appreciate it!! so,if student model is 7b,should be 56G,84G in total ,yet 2 *A800 (80G) doesn't seem enough,met with oom. batch_size is 4.

Do you OOM on the first step? If you reduce batch size and increase gradient accumulation does it work?

**Comment by ebsmothers** [comment]
Hi @chuangzhidan just getting back to this. I hacked in Llama 3.1 8B as the student model just to get a sense of how much memory is used with 8B student and 8B teacher. With [this config](https://gist.github.com/ebsmothers/7a6ed1240cbf274c040a62a22ddbc7e2) I found peak memory under 40 GiB when running on 2 A100s. So not sure why you're seeing OOMs. Maybe if you share your config we can help debug further though

**Comment by ebsmothers** [comment]
Hi @chuangzhidan do you still need help here? I am gonna close this issue, but if you're still stuck here please reopen and we can help debug
